### ISSUE TITLE
AccessViolationException on memberwise clone

### ISSUE THREAD

**Issue by detoxhby** [dadhi/FastExpressionCompiler#188]
**Environment**:
 - .NET Framework 4.6.2
 - System.Runtime(.Extensions): 4.3.1

**Description**:
After upgrading FEC from 1.7.1 to 2.0.0 (master), the memberwise based clone dies when trying to access fields with regular types.

```csharp
// this is the object's class being cloned
public class Role
{
        public Guid Id { get; set; } = Guid.NewGuid();

        // this is the member which contains data "UserManager" seen on the picture
        public string InternalName { get; set; }
}
```

```csharp
// this is called just before the field clone seen on the picture below
var clonedObject = MemberwiseCloneMethodInfo.Invoke(obj, null);
```
![image](https://user-images.githubusercontent.com/1679462/55750205-1699d600-5a43-11e9-8722-ebd2fe53d616.png)

**Problem**: the cloned string value is different than the original (seen as random chars on picture) for some reasons after the memberwise clone call. It does not happen with vanilla Expression.Compile or previous version.

**Remarks**
Please tell me what kind of contextual information you need to better repro this problem.

**Comment by dadhi** [comment]
Could you create a failing test in IssuesTest project and send a PR?

**Comment by detoxhby** [comment]
I'll have to extract out huge amount of nested (pre-)process code for this to be a simple repro, but will try to in the next days! :)

**Comment by detoxhby** [comment]
I traced the whole execution path and cannot determine the relation between FEC upgrade and this problem. Seemingly there is no connection as FEC were removed completely and replaced by vanilla Compile...still, if I have the source code of the v2 in the project, it does this mysteriously.

As I don't have enough time currently to determine the real cause I'm closing this issue but encourage anyone to try to reproudce this as there should be something related to FEC internally.

Thanks!
